website/pages/docs/plugins/sources/aws/event-based-sync.mdx

```diff
@@ -0,0 +1,92 @@
+import { Callout } from 'nextra-theme-docs';
+
+<Callout type="info">
+
+This feature is currently in a closed preview. Sign up for access using this [form](https://cloudquery.typeform.com/to/Dvdn6P2u)
+
+</Callout>
+
+# Event-Based Sync
+
+AWS CloudTrail enables users to get an audit log of events occurring within their account. By subscribing to a stream of AWS CloudTrail events in a Kinesis Data stream CloudQuery can trigger selective syncs to update just the singular resource that had a configuration change.
+
+
+## Supported Services and Events
+Service | Event 
+-|-|
+ec2.amazonaws.com | AttachInternetGateway
+ec2.amazonaws.com | AuthorizeSecurityGroupEgress
+ec2.amazonaws.com | AuthorizeSecurityGroupIngress
+ec2.amazonaws.com | CreateInternetGateway
+ec2.amazonaws.com | CreateSecurityGroup
+ec2.amazonaws.com | CreateSubnet
+ec2.amazonaws.com | CreateTags
+ec2.amazonaws.com | CreateVpc
+ec2.amazonaws.com | DeleteTags
+ec2.amazonaws.com | DetachInternetGateway
+ec2.amazonaws.com | ModifySubnetAttribute
+ec2.amazonaws.com | RevokeSecurityGroupEgress
+ec2.amazonaws.com | RevokeSecurityGroupIngress
+ec2.amazonaws.com | RunInstances
+iam.amazonaws.com | CreateGroup
+iam.amazonaws.com | CreateRole
+iam.amazonaws.com | CreateUser
+rds.amazonaws.com | CreateDBCluster
+rds.amazonaws.com | CreateDBInstance
+
+
+## Configuration
+
+1. Configure an AWS CloudTrail Trail to send management events to a Kinesis Data Stream via CloudWatch Logs. The most straight-forward way to do this is to use the CloudFormation template provided by CloudQuery.
+
+```bash
+aws cloudformation deploy --template-file ./streaming-deployment.yml --stack-name <STACK-NAME> --capabilities CAPABILITY_IAM --disable-rollback --region <DESIRED-REGION>
+```
+
+2. Copy the ARN of the Kinesis stream. If you used the CloudFormation template you can run the following command:
+```bash
+aws cloudformation describe-stacks --stack-name <STACK-NAME> --query "Stacks[].Outputs" --region <DESIRED-REGION>
+```
+
+3. Define a `config.yml` file like the one below
+
+``` yaml
+kind: source
+spec:
+  name: "aws-event-based"
+  registry: "local"
+  path: <PATH/TO/BINARY>
+  tables:
+    - aws_ec2_instances
+    - aws_ec2_internet_gateways
+    - aws_ec2_security_groups
+    - aws_ec2_subnets
+    - aws_ec2_vpcs
+    - aws_ecs_cluster_tasks
+    - aws_iam_groups
+    - aws_iam_roles
+    - aws_iam_users
+    - aws_rds_instances
+  destinations: ["postgresql"]
+  skip_tables:
```

Comment:
Is skipping these tables required? What happens if we remove this part of the configuration?

Reply:
Since we only support `create` events right now and not modify, those records would always be nil/empty but take extra API calls, which adds to latency.... It is a tradeoff between latency and actionable data

Reply:
Can you add a bit more context? Why those records will always by `nil` or empty? And why would we need to do extra API calls for them? If we know these are not supported in advance, could we skip them in the code?

Reply:
Currently we are only supporting triggering syncing on top level tables... These are nested tables... We could just remove the skip tables and fix the performance issues if they arise... this might have been a pre mature optimization

Reply:
Is there a reason the plugin can't ignore what it doesn't support?

Reply:
I don't think we want to skip all dependent tables.

What do you mean ignore?

Reply:
> I don't think we want to skip all dependent tables.

We can if this is only an event based sync

> What do you mean ignore?

Maybe I'm missing something, but the plugin knows which tables it supports, so I'm not sure why the user needs to configure those in `skip_tables`.

Reply:
It is a balance between performance and syncing data... We do not know how the user will be deploying this. If a user is syncing from a stream and catching up to real time, a user might not want to skip these tables as time has elapsed and there might be valuable usage data returned... So I don't think we can change any of the code to optimize for these different syncing patterns

Reply:
So I'm probably missing something because I thought we don't support these tables as a part of the event based sync.